components/tikv_util/src/metrics/mod.rs

```diff
@@ -59,7 +77,20 @@ pub fn dump_to(w: &mut impl Write, should_simplify: bool) {
             MetricType::COUNTER => {
                 metrics.retain(|m| m.get_counter().get_value() > 0.0);
             }
-            MetricType::HISTOGRAM => metrics.retain(|m| m.get_histogram().get_sample_count() > 0),
+            MetricType::HISTOGRAM => {
+                let threshold = if compact_policy == MetricsCompactPolicy::Lossless {
+                    0
+                } else {
+                    // only retain histogram that the sample count > 0.01 * max_sample_count
```

Comment:
This can be a big problem for analyzing tail latency.

Reply:
Yes. So this level should be only used when metrics data size is significant.

Reply:
I suggest to keep the highest data point at least.